WooCommerce/Classes/ViewRelated/Blaze/BudgetSetting/BlazeBudgetSettingView.swift

```diff
@@ -0,0 +1,301 @@
+        VStack(alignment: .leading, spacing: Layout.sectionContentSpacing) {
+            Divider()
+
+            // Duration title
+            Text(Localization.duration)
+                .secondaryBodyStyle()
+                .padding(.horizontal, Layout.contentPadding)
+                .padding(.top, Layout.sectionContentSpacing)
+
+            // Formatted duration
+            HStack {
+                Text(viewModel.formattedDateRange)
+                    .bold()
+                    .bodyStyle()
+                Text("·")
+                Text(Localization.edit)
+                    .foregroundColor(.accentColor)
+                    .bodyStyle()
+            }
+            .padding(.horizontal, Layout.contentPadding)
+            .onTapGesture {
+                showingDurationSetting = true
+            }
+
+            // CTA to confirm all settings
+            Button(Localization.update) {
+                viewModel.confirmSettings()
+                dismiss()
+            }
+            .buttonStyle(PrimaryButtonStyle())
+            .padding([.horizontal, .bottom], Layout.contentPadding)
+            .padding(.top, Layout.sectionContentSpacing)
+        }
+        .background(Color(.systemBackground))
+    }
+
+    var impressionInfoView: some View {
+        NavigationView {
+            ScrollView {
+                Text(Localization.impressionInfo)
+                    .bodyStyle()
+                    .padding(Layout.contentPadding)
+            }
+            .navigationTitle(Localization.title)
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button(Localization.done) {
+                        showingImpressionInfo = false
+                    }
+                }
+            }
+        }
+        .padding(.top)
+    }
+
+    var durationSettingView: some View {
+        ScrollView {
+            // Title
+            Text(Localization.setDuration)
+                .headlineStyle()
+                .padding(.horizontal, Layout.contentPadding)
+                .padding(.vertical, Layout.sectionSpacing)
+
+            Spacer()
+
+            // Duration slider
+            VStack(spacing: Layout.sectionContentSpacing) {
+                Text(viewModel.formattedDayCount)
+                    .fontWeight(.semibold)
+                    .bodyStyle()
+
+                Slider(value: $viewModel.dayCount,
+                       in: viewModel.dayCountSliderRange,
+                       step: Double(BlazeBudgetSettingViewModel.Constants.dayCountSliderStep))
+            }
+            .padding(Layout.contentPadding)
+
+            // Start date picker
+            VStack {
+                HStack {
+                    Text(Localization.starts)
+                        .bodyStyle()
+
+                    Spacer()
+
+                    DatePicker(selection: $viewModel.startDate, displayedComponents: [.date]) {
```

Comment:
Can this be made to prevent selecting past dates? I think it makes no sense to make a campaign starting in the past.

Reply:
Nice catch! Added the date range and sneaked in a layout fix in b09338a4712d83c6d31b058fc0784b9530cae0cc.